### DISCUSSION TITLE
Downstream partition (daily) depending on an less granular upstream partition (monthly)

### DISCUSSION THREAD

**Discussion by bijzz** [dagster-io/dagster#14113] [Q&A]
In partition dependencies https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#partition-dependencies the example describes a case where a downstream asset is less granular partitioned (daily) than an upstream asset (hourly). Does this also work vice versa? 

In my case, i have a daily partitioned table that depends on monthly partitioned table. Additionally, in this case i am also interested in an PartitionMapping with an offset such that the daily table depends on the previous months (-1) or even pre-previous (-2) months aggregations. Example:

![Unbenannt-2023-05-05-1206](https://user-images.githubusercontent.com/1084828/236433150-461fd7df-bd1d-4cf4-8119-ddb35490baaa.png)

Is this possible? Also, if i need to use https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#partitionmappings how can i set this setting for assets imported via DBT and `load_assets_from_dbt_project`?

**Comment by bijzz** [comment]
Just found a more detailed documentation at https://docs.dagster.io/_apidocs/partitions#dagster.TimeWindowPartitionMapping.

Judging from this it should be possible that an upstream is more coarse than a downstream asset. Unfortunately, the start_offset and end_offset can only be applied to identically partitioned assets. Still, i tried the following:

monthly upstream: materialized 2023-05-01 (i shifted my monthly partitioning into the future with end_offset=1)
daily upstream: materialized 2023-05-04
daily downstream: want to materialize 2023-05-04, but dagster still complains:

![image](https://user-images.githubusercontent.com/1084828/236522104-359331ce-6d90-4a33-a0d0-3c9dae2e282e.png)

![image](https://user-images.githubusercontent.com/1084828/236522284-3b1931ad-3d6d-4ead-b599-155d99acb966.png)

Should this not work according to the documentation? And will it be possible in the future to define an offset here as well to allow the daily downstream to depend on an earlier month upstream?

**Reply by sryza** [reply]
> Should this not work according to the documentation?

Hmm - what happens if you try to run the materialization? If you look at `monthly_table`, does it have a partition for `2023-05`? It's possible this is a frontend error.

> will it be possible in the future to define an offset here as well to allow the daily downstream to depend on an earlier month upstream?

We'd like it to be possible, but it's not on our near term roadmap for bandwidth reasons. If you'd be up for filing an issue, that could bump the eventual priority.  We'd also welcome a contribution.

**Reply by bijzz** [reply]
Indeed it seems to be a frontend error. I activated the AutoMaterialization Sensor and the daily downstream table kicked off for the latest business date.

For allowing offsets i filed an issue https://github.com/dagster-io/dagster/issues/14152.
